builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -484,6 +483,16 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("Invalid network interfaces to set: %#v", networkInterfaces)
 	}
 
+	ip, err := vm.WaitForIP(context.TODO())
```

Comment:
@tkak shouldn't we be doing this earlier in the method? If we move it to before `collector.RetrieveOne` we may get better results.  The thought that I have is that we are setting the network interfaces from `Guest.Net` and the data may not be populated until `vm.WaitForIP` returns.  I am pushing a change to a branch on my repo, and maybe someone can test.

@ryanl-ee have you tested this change on this pull?  I am still looking for a test bed unfortunately, but I will make the change that I mentioned to @tkak